src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -115,13 +123,21 @@ public bool ExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionConte
         }
 
         private void CodeCleanupOrFormat(
-            FormatDocumentCommandArgs args, Document document,
+            FormatDocumentCommandArgs args, Document document, ICodeCleanupService codeCleanupService,
             IProgressTracker progressTracker, CancellationToken cancellationToken)
         {
-            var codeCleanupService = document.GetLanguageService<ICodeCleanupService>();
+            var workspace = document.Project.Solution.Workspace;
+            var isFeatureTurnedOnThroughABTest = TurnOnCodeCleanupForGroupBIfInABTest(document, workspace);
 
-            var docOptions = document.GetOptionsAsync(cancellationToken).WaitAndGetResult(cancellationToken);
-            if (docOptions.GetOption(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting))
+            // if feature is turned on through AB test, we need to show the Gold bar even if they set NeverShowCodeCleanupInfoBarAgain == true before
+            if (isFeatureTurnedOnThroughABTest ||
+                !workspace.Options.GetOption(CodeCleanupOptions.NeverShowCodeCleanupInfoBarAgain, document.Project.Language))
```

Comment:
>!workspace.Options.GetOption(CodeCleanupOptions.NeverShowCodeCleanupInfoBarAgain, document.Project.Language) [](start = 16, length = 108)

Consider making this the first part of the `if`, and then moving `TurnOnCodeCleanupForGroupBIfInABTest` to the second part, so we don't even query the experimentation service if they've already turned off the bar.

Reply:
@333fred  I talked with @kuhlenh , and she said if a user previously turned off the bar and got assigned into the experiment group this time, we still want the gold bar to show.

Reply:
isn't this will make this person to see goldbar in every new VS session? are we believe number of people who meet this criteria is large enough to influence AB testing result?. otherwise, I would just remove this.